presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTaskFactory.java

```diff
@@ -165,4 +181,62 @@ public RemoteTask createRemoteTask(Session session,
                 stats,
                 isBinaryTransportEnabled);
     }
+
+    @Override
+    public void removeRemoteSource(TaskId taskId, TaskId remoteSourceTaskId, Consumer<PrestoException> onFailure)
+    {
+        // TODO: Check whether onFailure callback is needed. Because this method is used for recoverable execution,
+        //  and if cleanup fails the query will fail.
+        URI remoteSourceUri = uriBuilderFrom(locationFactory.createLocalTaskLocation(taskId))
+                .appendPath("remote-source")
+                .appendPath(remoteSourceTaskId.toString())
+                .build();
+
+        Request request = prepareDelete()
+                .setUri(remoteSourceUri)
+                .build();
+        RequestErrorTracker errorTracker = new RequestErrorTracker(
+                taskId,
+                remoteSourceUri,
+                maxErrorDuration,
+                errorScheduledExecutor,
+                "Remove exchange remote source");
+
+        executeDestroyExchangeSourceRequest(errorTracker, request, onFailure);
+    }
+
+    private void executeDestroyExchangeSourceRequest(RequestErrorTracker errorTracker, Request request, Consumer<PrestoException> onFailure)
+    {
+        errorTracker.startRequest();
+
+        ResponseHandler responseHandler;
```

Comment:
The REST method returns `void`, empty output. Use the `StatusResponseHandler` there.

Make sure that response status is `200`.

Reply:
`StatusResponseHandler` doesn't have wrapped smile encoding adapted response handler yet :(

Reply:
I saw it was changed to StatusResponseHandler. Resolving this.